src/app/cli/src/coda.ml

```diff
@@ -632,9 +632,59 @@ let daemon logger =
            Auxiliary_database.External_transition_database.create ~logger
              external_transition_database_dir
          in
+         (* log terminated child processes *)
+         let pids = Child_processes.Termination.create_pid_set () in
+         let rec terminated_child_loop () =
+           match
+             try Unix.wait_nohang `Any
```

Comment:
How does this interact with the other calls to `Process.wait` in our codebase?

Reply:
I think it will be OK:

  - If there's a pending `wait` elsewhere, this call should return with `None`, and have no effect on the other `wait`
  - if this call returns `Some`, meaning the process has terminated, the other `wait` should
    return with the same code it would have without this call

At least, that's what I understand from reading `man wait`